### ISSUE TITLE
Android ScreenStack Screens require ScreenStackHeaderConfig child to change orientation and status bar props

### ISSUE THREAD

**Issue by scottmas** [software-mansion/react-native-screens#1310]
## Description
On Android only, the `Screen` props `screenOrientation` and `statusBar` (e.g. `statusBarHidden`) do nothing when the `Screen` is a child of a `ScreenStack`.

## Steps To Reproduce
1. Create a stock RnDiffApp `npx @react-native-community/cli init RnDiffApp`
2. Add react native screens `yarn add react-native-screens@latest` (currently 3.11.0)
3. Make your index.js file look like so: 
```javascript
const React = require('react');
const {AppRegistry, Text} = require('react-native');
const {ScreenStack, Screen, ScreenContainer} = require('react-native-screens');

/** This does NOT work **/
AppRegistry.registerComponent('RnDiffApp', () => () => (
  <ScreenStack>
    <Screen screenOrientation={'landscape'} statusBarHidden>
      <Text style={{paddingTop: 100}}>Status bar and orientation have not changed :(</Text>
    </Screen>
  </ScreenStack>
));

/** This works **/
// AppRegistry.registerComponent('RnDiffApp', () => () => (
//   <ScreenContainer>
//     <Screen activityState={2} screenOrientation={'landscape'} statusBarHidden>
//      <Text style={{paddingTop: 100}}>Status bar and orientation HAVE changed :)</Text>
//     </Screen>
//   </ScreenContainer>
// ));
```
4. Build and install the app

### Expected behavior

The status bar and orientation props should work on Android with a root ScreenStack

### Actual behavior
 
They don't work

## Reproduction
See above on steps to reproduce. No way that I can tell to create a snack since I'm using the react-native-screens primitives directly. 

## Platform

- [ ] iOS
- [X] Android
- [ ] Web
- [ ] Windows
- [ ] tvOS

**Comment by scottmas** [comment]
Interestingly, the following DOES work:
```javascript

<ScreenStack>
    <Screen screenOrientation={'landscape'} statusBarHidden>
      <ScreenContainer>
        <Screen activityState={2}>
          <Text style={{paddingTop: 100}}>Status bar and orientation have changed :)</Text>
        </Screen>
      </ScreenContainer>
    </Screen>
  </ScreenStack>
```

**Comment by WoLewicki** [comment]
On `Android`, when setting initial props connected to `statusBar` and `navigationBar`, you need `fragment` to get the `activity` (https://github.com/software-mansion/react-native-screens/blob/b22efc187d41ee4a95ad82321b68459ac10c8e8e/android/src/main/java/com/swmansion/rnscreens/Screen.kt#L252) , which sometimes is not there yet, so these props are evaluated again when updating `HeaderConfig` (https://github.com/software-mansion/react-native-screens/blob/3.10.2/android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.kt#L150), where we have `activity` provided. So in order for this to work for you, you should do it like that: 
```tsx
    <ScreenStack>
      <Screen screenOrientation={"landscape"} statusBarHidden >
        <ScreenStackHeaderConfig />
        <Text style={{paddingTop: 100}}>Status bar and orientation have not changed :(</Text>
      </Screen>
    </ScreenStack>
 ```

Can you check if it works then? To confirm that it works like that, you can try and set a prop on `Screen` without `ScreenStackHeaderConfig` after a timeout, e.g. 500ms, since then the `fragment` will be there and it should work normally. Can I help you more with this?

**Comment by scottmas** [comment]
That does indeed fix it for Android. Could there be a way to "queue" the modifications to ScreenWindowTraits for when the fragment mounts?

Also, while debugging this, I also noticed an issue on iOS. When I push a change from portrait to landscape, there's a 1 second delay (only on iOS) and I briefly see the screen in portrait mode. For example, when I go from this state:
```javascript
  <ScreenStack>
    <Screen screenOrientation={'portrait'}>
      <Text style={{paddingTop: 100}}>Portrait orientation</Text>
    </Screen>
  </ScreenStack>
```

to this one

```javascript
  <ScreenStack>
    <Screen screenOrientation={'portrait'}>
      <Text style={{paddingTop: 100}}>Portrait orientation</Text>
    </Screen>
    <Screen screenOrientation={'landscape'}>
      <Text style={{paddingTop: 100}}>Landscape orientation, but with a flash of portrait orientation</Text>
    </Screen>
  </ScreenStack>
```
The screen saying "Landscape orientation" has a roughly one second flash in portrait orientation before it switches to landscape.

So basically, it just seems like setting orientation and status bar props on stacks is kind of problematic and maybe not to be recommended...

**Comment by WoLewicki** [comment]
Does this one-second delay differ from how it happens in native apps? I am not sure if we do anything specific for resolving the orientation, so it seems weird that it would happen just here. All traits are resolved in `viewWillAppear`: https://github.com/software-mansion/react-native-screens/blob/08464d2b9ac175a2c5f005547d3afc4c9863a1e4/ios/RNSScreen.m#L594 which seems the earliest possible place to do it. Am I missing something?

**Comment by scottmas** [comment]
I'm not familiar with native programming, so I don't know the answer to that. I'm sorry! What I do know though is that the transition is much snappier (instant) when I set the landscape `screenOrientation` prop on on a `ScreenContainer` rather than a `ScreenStack` (aka when I use a tab navigator rather than a stack navigator). However, when I do a minimal test case to compare the amount of time the screen has the "incorrect" orientation, it appears to be about the same whether a stack or a tab navigator. 

But this is a tangent that's neither here nor there. I'll open a new issue with this issue of the incorrect screen dimensions.

What do you think should be done about the original issue of the Stack Navigator? Are you just going to tell library authors that they should always render a ScreenStackHeaderConfig whenever they render a stack navigator? Or is there a way to not have that requirement? In either case, there's not much more I can do on this ticket since I'm an absolute novice in native programming.

**Comment by WoLewicki** [comment]
I think it would be best to change the issue name to what it concerns now and add minimal reproduction code where we can see those problems and how it differs between `ScreenContainer` and `ScreenStack`. About the necessity of `ScreenStackHeaderConfig`, I think we will stick to this for now since I have no better idea how to deal with it.

**Comment by scottmas** [comment]
Reposting the repro code here:

```javascript
const React = require('react');
const {AppRegistry, Text} = require('react-native');
const {ScreenStack, Screen, ScreenContainer} = require('react-native-screens');

/** This does NOT work **/
AppRegistry.registerComponent('RnDiffApp', () => () => (
  <ScreenStack>
    <Screen screenOrientation={'landscape'} statusBarHidden>
      <Text style={{paddingTop: 100}}>Status bar and orientation have not changed :(</Text>
    </Screen>
  </ScreenStack>
));

/** But it works on ScreenContainer **/
// AppRegistry.registerComponent('RnDiffApp', () => () => (
//   <ScreenContainer>
//     <Screen activityState={2} screenOrientation={'landscape'} statusBarHidden>
//      <Text style={{paddingTop: 100}}>Status bar and orientation HAVE changed :)</Text>
//     </Screen>
//   </ScreenContainer>
// ));
```

**Comment by WoLewicki** [comment]
Ok so I think we can close this issue since the problem it concerns is resolved by adding `ScreenStackHeaderConfig`. Feel free to comment here if something is wrong and we can always reopen it then.
